### PR TITLE
[rpc] expose consensus internal data via rpc

### PR DESCRIPTION
**Pull request by LeoHChen**
This is used to debug consensus/viewchange state.

Signed-off-by: Leo Chen <leo@harmony.one>

### CODE REVIEW

**Comment by LeoHChen** [comment]
image


Example of watchdog output on STN.

File: node/api.go [hunk 1]
```diff
@@ -126,3 +127,21 @@ func (node *Node) GetConsensusViewChangingID() uint64 {
...
+
+// GetConsensusInternal returns consensus internal data
+func (node *Node) GetConsensusInternal() rpc_common.ConsensusInternal {
+	return rpc_common.ConsensusInternal{
+		ViewID:        node.GetConsensusCurViewID(),
+		VCID:          node.GetConsensusViewChangingID(),
+		Mode:          node.GetConsensusMode(),
+		Phase:         node.GetConsensusPhase(),
+		BlockNum:      node.GetConsensusBlockNum(),
+		InSync:        !node.IsOutOfSync(node.Blockchain()),
```
Comment (Daniel-VDM) [inline node/api.go:144]: This locks the syncing service, prob best to not put this in the node metadata as its called quite frequently.

File: hmy/hmy.go [hunk 2]
```diff
@@ -190,6 +194,7 @@ func (hmy *Harmony) GetNodeMetadata() commonRPC.NodeMetadata {
 		Archival:       cfg.GetArchival(),
 		NodeBootTime:   hmy.NodeAPI.GetNodeBootTime(),
 		PeerID:         nodeconfig.GetPeerID(),
+		Consensus:      consensusInternal,
```
Comment (Daniel-VDM) [inline hmy/hmy.go:197]: let's not put this in metadata, reason is in a different comment.

File: rpc/common/types.go [hunk 3]
```diff
@@ -37,6 +37,17 @@ type C struct {
...
 
+// ConsensusInternal captures consensus internal data
+type ConsensusInternal struct {
+	ViewID        uint64 `json:"viewid"`
+	VCID          uint64 `json:"vcid"`
+	Mode          string `json:"mode"`
+	Phase         string `json:"phase"`
+	BlockNum      uint64 `json:"blocknum"`
+	InSync        bool   `json:"insync"`
+	ConsensusTime int64  `json:"finality"`
```
Comment (Daniel-VDM) [inline rpc/common/types.go:48]: `vcid` is not clear IMO. Maybe also stick to [kebab case](https://wiki.c2.com/?KebabCase) for the unmarshalled names if we decide to keep it in node metadata. Or use snake case.

File: node/api.go (hunk 1 above)
Comment (Daniel-VDM) [reply node/api.go:144]: Suggestion is to put it as a separate RPC.

**Review by Daniel-VDM — CHANGES_REQUESTED** [review]
Some concerns about putting it in node metadata & some naming nits

File: node/api.go (hunk 1 above)
Comment (LeoHChen) [reply node/api.go:144]: got it, will remove it.

File: hmy/hmy.go (hunk 2 above)
Comment (LeoHChen) [reply hmy/hmy.go:197]: I'll remove the IsSync variable then.

**Review by Daniel-VDM — APPROVED** [review]

**Pull request merged**